src/main/java/jdbchomework/dao/hibernate/AbstractHibDao.java

```diff
@@ -0,0 +1,122 @@
+    public static final String CANNOT_CONNECT_TO_DB = "Cannot connect to DB";
+    private org.slf4j.Logger log = LoggerFactory.getLogger(AbstractHibDao.class);
+    protected SessionFactory sessionFactory;
+    private String entityName;
+    private Class<T> clazz;
+
+    public AbstractHibDao(String entityName, SessionFactory sessionFactory, Class<T> clazz) {
+        this.entityName = entityName;
+        this.sessionFactory = sessionFactory;
+        this.clazz = clazz;
+
+    }
+
+    @Override
+    public void add(T toAdd) {
+        Transaction transaction = null;
+
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(toAdd);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error(CANNOT_CONNECT_TO_DB, e);
+            throw new RuntimeException(CANNOT_CONNECT_TO_DB, e);
+        }
+
+    }
+
+    @Override
+    public List<T> getAll() {
+        List<T> result = new ArrayList<T>();
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            result = session.createQuery("from " + entityName).list();
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error(CANNOT_CONNECT_TO_DB, e);
+            throw new RuntimeException(CANNOT_CONNECT_TO_DB, e);
+        }
+        return result;
+    }
+
+    @Override
+    public T getById(long id) {
+        T result = null;
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            result = session.get(clazz, id);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error(CANNOT_CONNECT_TO_DB, e);
+            throw new RuntimeException(CANNOT_CONNECT_TO_DB, e);
+        }
+        return result;
+    }
+
+    @Override
+    public boolean deleteById(long id) {
+        Transaction transaction = null;
+        boolean res = false;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            T t = getById(id);
+            session.delete(t);
+            transaction.commit();
+            res = true;
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error(CANNOT_CONNECT_TO_DB, e);
+            throw new RuntimeException(CANNOT_CONNECT_TO_DB, e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)